jcl/src/java.base/share/classes/java/lang/invoke/MethodHandles.java

```diff
@@ -497,7 +506,7 @@ private void checkAccess(Class<?> targetClass, String name, int memberModifiers,
 				}
 			} else {
 				/* default (package access) */
-				if ((PACKAGE == (accessMode & PACKAGE)) && isSamePackage(accessClass, targetClass)){
+				if ((PACKAGE == (accessMode & PACKAGE)) && isSamePackage(accessClass, referenceClass)){
```

Comment:
I am not too sure whether it is correct to change to referenceClass.

Reply:
`referenceClass` must be a subtype of `definingClass` in order to inherit the method.